core/App/ehsm_provider.cpp

```diff
@@ -810,4 +811,46 @@ ehsm_status_t ExportDataKey(ehsm_keyblob_t *cmk,
         return EH_OK;
 }
 
+std::string create_apikey()
+{
+    sgx_status_t sgxStatus = SGX_ERROR_UNEXPECTED;
+    sgx_status_t ret = SGX_ERROR_UNEXPECTED;
+    
+    uint32_t apikey_len;
+
+    // calculate length
+    ret = enclave_create_apikey(g_enclave_id,
+                                &sgxStatus,
+                                NULL,
+                                0,
+                                &apikey_len);
+    if (ret != SGX_SUCCESS || sgxStatus != SGX_SUCCESS){
+        printf("error: calculate apikey length faild (%d)(%d).\n", ret, sgxStatus);
+        return "";
+    }
+    
+    uint8_t p_apikey[apikey_len];
+    // create apikey
+    if (p_apikey == NULL) {
+        printf("error: apikey malloc faild.\n");
+        return "";
+    }
+    ret = enclave_create_apikey(g_enclave_id,
```

Comment:
Let's define the length as 32B by default, so that the function is not required to call twice.